neptune/new/internal/operation_processors/async_operation_processor.py

```diff
@@ -128,29 +164,17 @@ def work(self) -> None:
                     return
                 self.process_batch(batch, version)
 
+        @Daemon.ConnectionRetryWrapper()
+        def process_loop(self, batch, version):
+            result = self._processor._backend.execute_operations(self._processor._run_uuid, batch)
+            self._processor._queue.ack(version)
+            for error in result:
+                _logger.error("Error occurred during asynchronous operation processing: %s", error)
+            return result
+
         def process_batch(self, batch: List[Operation], version: int) -> None:
             # TODO: Handle Metadata errors
-            for retry in range(0, self.RETRIES):
-                try:
-                    result = self._processor._backend.execute_operations(self._processor._run_uuid, batch)
-                    self._processor._queue.ack(version)
-                    for error in result:
-                        _logger.error("Error occurred during asynchronous operation processing: %s", error)
-                    break
-                except NeptuneConnectionLostException:
-                    if retry >= self.RETRIES - 1:
-                        click.echo("Experiencing connection interruptions. Killing Neptune asynchronous thread. "
-                                   "All data is safe on disk.",
-                                   sys.stderr)
-                        raise
-                    click.echo("Experiencing connection interruptions. Reestablishing communication with Neptune.",
-                               sys.stderr)
-                    sleep(self.RETRY_WAIT)
-                except Exception:
-                    click.echo("Unexpected error occurred. Killing Neptune asynchronous thread. "
-                               "All data is safe on disk.",
-                               sys.stderr)
-                    raise
+            self.process_loop(batch, version)
```

Comment:
So now we have method calling another method without changing anything. Could we wrap process_batch with ConnectionRetryWrapper and have only one method?

Reply:
I got lost in the refactoring a little bit sorry. Will do that.